pypeit/core/wavecal/autoid.py

```diff
@@ -1772,12 +1772,14 @@ def run_brute(self, min_nlines=10):
             # Print preliminary report
             good_fit[slit] = self.report_prelim(slit, best_patt_dict, best_final_fit)
 
-        # Now that all slits have been inspected, cross match to generate a
+        # Now that all slits have been inspected, cross match (if there are bad fit) to generate a
         # list of all lines in every slit, and refit all spectra
-        obad_slits = self.cross_match(good_fit, self._det_weak) if self._ok_mask.size > 1 else np.array([])
-        if obad_slits.size > 1:
+        # in self.cross_match() good fits are cross correlate with each other, so we need to have at least 2 good fits
+        if np.where(good_fit[self._ok_mask])[0].size > 1 and np.any(np.logical_not(good_fit[self._ok_mask])):
```

Comment:
Should this be `np.where(good_fit[self._ok_mask])[0].size >= 1`? I think there only needs to be one good fit in order to continue?

Reply:
@rcooke-ast as I wrote in the comment in line 1777, `self.cross_match()`  cross correlates the good fit with each other before cross matching with the bad fits, therefore there needs to be at least 2 good fits to do that.

Reply:
Ahh yes, thanks for the explanation - all looks good to me!